lms/static/scripts/frontend_apps/components/URLPicker.js

```diff
@@ -19,13 +24,28 @@ export default function URLPicker({ onCancel, onSelectURL }) {
   const input = useRef(/** @type {HTMLInputElement|null} */ (null));
   const form = useRef(/** @type {HTMLFormElement|null} */ (null));
 
+  // Holds an error message corresponding to client-side validation of the
+  // input field
+  const [error, setError] = useState(/** @type {string|null} */ (null));
+
   /** @param {Event} event */
   const submit = event => {
     event.preventDefault();
-    if (form.current.checkValidity()) {
-      onSelectURL(input.current.value);
-    } else {
-      form.current.reportValidity();
+    try {
+      const url = new URL(input.current.value);
+      if (!url.protocol.startsWith('http')) {
+        if (url.protocol.startsWith('file')) {
+          setError(
+            'URLS that start with "file" are files on your own computer. Please use a URL that starts with "http" or "https".'
+          );
+        } else {
+          setError('Please use a URL that starts with "http" or "https"');
+        }
+      } else {
+        onSelectURL(input.current.value);
+      }
+    } catch (e) {
+      setError('Please enter a URL');
```

Comment:
I wonder if it may help to include a valid example here in case the user enters something that is not quite a URL (`example.com/dummy.pdf`) and isn't sure what they did wrong.

Reply:
That's a nice idea! I'm adding that.